Kerberos.NET/Client/Transport/SocketPool.cs

```diff
@@ -100,6 +100,7 @@ public void Poll(TimeSpan window)
                 if (Queue.TryDequeue(out socket))
                 {
                     socket.Free();
+                    Interlocked.Decrement(ref activeConnections);
```

Comment:
Am I understanding this correctly? When the pool checks if it needs to clean up any unused connections it'll free the socket but leave the active connection count incremented, so the next time it tries to get a new socket it hits the upper limit?